src/kvilib/system/KviEnvironment.cpp

```diff
@@ -33,14 +33,14 @@ namespace KviEnvironment
 
 #if !defined(COMPILE_ON_WINDOWS) && !defined(COMPILE_ON_MINGW)
 
-#ifdef HAVE_SETENV
-	bool setVariable(const char * name, const char * value)
+	bool setVariable(QString szName, QString szValue)
 	{
+		const char * name = szName.toLocal8Bit().data();
+		const char * value = szValue.toLocal8Bit().data();
```

Comment:
`QString::toLocal8Bit` returns a temporary `QByteArray`, so `name` and `value` are pointing to invalid locations in memory.  Consider `auto name = szName.toLocal8Bit(); ... setenv(name.data(), value.data(), ...);`